### ISSUE TITLE
The argument "subscription_id" is required, but no definition was found

### ISSUE THREAD

**Issue by shess-macu** [hashicorp/terraform-provider-azurerm#27154]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.9.4

### AzureRM Provider Version

4.0.0

### Affected Resource(s)/Data Source(s)

azurerm provider

### Terraform Configuration Files

```hcl
provider "azurerm" {
  features {}
}
```

### Debug Output/Panic Output

```shell
Error: Missing required argument
│
│   on base.tf line 1, in provider "azurerm":
│    1: provider "azurerm" {
│
│ The argument "subscription_id" is required, but no definition was found.
```

### Expected Behaviour

subscription_id should not be required

### Steps to Reproduce

create a basic terraform config with azurerm provider using version 4.0.0. Run `terraform init` and then `terraform validate`

**Comment by shess-macu** [comment]
I was able to confirm all previous versions of the azurerm provider work as expected. Including version 3.116.0.

**Comment by shess-macu** [comment]
I was able to find the documentation on this. Looks like this is expected behavior. Which is unfortunate.
https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/guides/4.0-upgrade-guide#specifying-subscription-id-is-now-mandatory

**Comment by stephybun** [comment]
Thanks for raising this issue @shess-macu!

As you've already found, this is one of the breaking changes we introduced in 4.0. Since this is neither a bug nor a feature request I'm going to go ahead and close this issue.

**Comment by LiamLeane** [comment]
This breaks a large number of CI scenarios where terraform doesn't, and should not, have subscription/tenant details available. I absolutely understand requiring it to be set for CLI auth when an apply is performed but not for a validation.

Additionally in cases where engineers intentionally do not have access to subscriptions and instead rely on automation via TFC/GHA/etc to plan their changes it means they can't perform a terraform validate without setting a dummy value.

This is extremely inconsistent with all the other providers. Are you really telling us we have to set an env var to 00000000-0000-0000-0000-000000000000 for the provider to be able to validate?

**Comment by manicminer** [comment]
@LiamLeane Thanks for reporting, that's definitely a bug. It should be possible to run `terraform validate` without providing authentication details. We'll issue a fix for this shortly.

**Comment by gustavonj** [comment]
@stephybun do you broke the terraform validate with this changes.

**Comment by gustavonj** [comment]
> @stephybun this change broke the terraform validate :( and CI pipelines

@stephybun this change broke the terraform validate :( and CI pipelines

**Comment by gustavonj** [comment]
> Thanks for raising this issue @shess-macu!
> 
> As you've already found, this is one of the breaking changes we introduced in 4.0. Since this is neither a bug nor a feature request I'm going to go ahead and close this issue.

@stephybun this change broke the terraform validate :( and CI pipelines

**Comment by manicminer** [comment]
@gustavonj This will be fixed in v4.0.1 (see #27178)

**Comment by shess-macu** [comment]
Thank you @manicminer !
